crates/search/src/buffer_search.rs

```diff
@@ -504,6 +504,7 @@ impl BufferSearchBar {
         let replacement_editor = cx.new_view(|cx| Editor::single_line(cx));
         cx.subscribe(&replacement_editor, Self::on_replacement_editor_event)
             .detach();
+        let search_options = SearchOptions::from_defaults(&EditorSettings::get_global(cx).search_defaults);
```

Comment:
Unlike project searches which are instantiated on every action call, buffer search is only created a few times; once for all code-related buffers:

https://github.com/zed-industries/zed/blob/8ee42d37423ac8734595522d680aa186b617ccf3/crates/zed/src/zed.rs#L582


that's why we need to do more here, to reflect the settings changes.
I believe, we can update any of the `deploy` (or `show`/`toggle`/etc. instead?) methods to do the same settings retrieval and an update.

Reply:
Thanks. I'll to look to see where we can update it. You know how I can watch for a change in the settings

Reply:
You could have another subscription like https://github.com/zed-industries/zed/blob/eecff753fbd5ab7eecc886122cd509cb45a97326/crates/outline_panel/src/outline_panel.rs#L656-L662